<launch>
   <arg name="use_ptgrey" default="false" />
   
   <!-- Camera Nodes -->
   <!-- Point Grey -->
   <group if="$(arg use_ptgrey)" ns="usb_cam">
      <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" />

      <node pkg="nodelet" type="nodelet" name="camera_nodelet"
          args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >
         <param name="frame_id" value="camera_link" />
         <param name="frame_rate" value="30" />
         <param name="format7_roi_width" value="1280" />
         <param name="format7_roi_height" value="960" />
         <param name="format7_x_offset" value="400" />
         <param name="format7_y_offset" value="296" />
      </node>

      <node pkg="nodelet" type="nodelet" name="image_proc_debayer"
         args="load image_proc/debayer camera_nodelet_manager">
      </node>
   </group>
   
   <!-- Common Webcam -->
   <group unless="$(arg use_ptgrey)" >
      <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
         <param name="video_device" value="/dev/video0" />
         <param name="image_width" value="640" />
         <param name="image_height" value="480" />
         <param name="pixel_format" value="yuyv" />
         <param name="camera_frame_id" value="usb_cam" />
         <param name="io_method" value="mmap"/>
      </node>
   </group>
   
   <!-- Tracker Nodes -->
   <include file="$(find vtec_tracker)/launch/tracker.launch"/>
</launch>